[pt1][quant] Make the default qconfig_dict

jianyuh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23139 [pt1][quant] Unit Test for the Dynamic Quantized Linear operator
* #24231 [pt1][quant] Add the default_weight_observer for the dynamic quantization path
* #24299 [pt1][quant] Remove the activation observer for default_qconfig
* **#24232 [pt1][quant] Make the default qconfig_dict**
* #23330 [pt1][quant] Fix the dimension mismatch issues when running the BERT model
* #23212 [pt1][quant] Add the type matching rule for qconfig_dict

As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

jianyuh committed:
[pt1][quant] Make the default qconfig_dict

As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

jianyuh committed:
Update on "[pt1][quant] Make the default qconfig_dict"


As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

jianyuh committed:
Update on "[pt1][quant] Make the default qconfig_dict"


As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

jianyuh committed:
Update on "[pt1][quant] Make the default qconfig_dict"


As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

jianyuh committed:
Update on "[pt1][quant] Make the default qconfig_dict"


As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

jianyuh committed:
Update on "[pt1][quant] Make the default qconfig_dict"


As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

jianyuh committed:
Update on "[pt1][quant] Make the default qconfig_dict"


As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)

raghuramank100 commented on torch/quantization/quantize.py:
```diff
@@ -221,7 +222,11 @@ def quantize(model, run_fn, run_args, mapping=DEFAULT_MODULE_MAPPING):
-def quantize_dynamic(model, qconfig_dict=None, mapping=DEFAULT_DYNAMIC_MODULE_MAPPING):
+DEFAULT_QCONFIG_DICT = {
+    nn.Linear : default_qconfig
+}
```
I see that default_qconfig is updated to remove activation observers in #24231

jianyuh committed:
Update on "[pt1][quant] Make the default qconfig_dict"


As suggested in https://github.com/pytorch/pytorch/pull/23128#discussion_r306650311, we will make the keys of default_qconfig_dict as `torch.nn.Linear`. That is, we will do the dynamic quantization on the `torch.nn.Linear` by default, if the user just specify `torch.quantize_dynamic(model)`.

Differential Revision: [D16781191](https://our.internmc.facebook.com/intern/diff/D16781191/)
